openc3/lib/openc3/api/cmd_api.rb

```diff
@@ -347,18 +357,19 @@ def cmd_implementation(method_name, *args, range_check:, hazardous_check:, raw:,
         'hazardous_check' => hazardous_check.to_s,
         'raw' => raw.to_s
       }
-      log_cmd = true
-      if packet["messages_disabled"]
-        log_cmd = false
-      else
+      if log_message.nil? # This means the default was used, no argument was passed
+        log_message = true # Default is true
+        # If the packet has the DISABLE_MESSAGES keyword then no messages by default
+        log_message = false if packet["messages_disabled"]
+        # Check if any of the parameters have DISABLE_MESSAGES
         cmd_params.each do |key, value|
           item = packet['items'].find { |item| item['name'] == key.to_s }
           if item['states'] && item['states'][value] && item['states'][value]["messages_disabled"]
-            log_cmd = false
+            log_message = false
           end
         end
       end
-      if log_cmd
+      if log_message
```

Comment:
Bonus to this is now you can FORCE output on a previously DISABLE_MESSAGES command or parameter.

Reply:
Very nice